provider/team_resource.go

```diff
@@ -33,6 +33,26 @@ func resourceTeam() *schema.Resource {
 				Type:     schema.TypeString,
 				Optional: true,
 			},
+			"memberships": {
```

Comment:
Is it correct that you need a separate `memberships` block per member that is added to the team?  If so, something like `membership` or `member` communicates that intention better IMO.  Is this a Terraform pattern or something?

Reply:
feels strange to me as well but it's a pattern we use. For example runbooks takes a bunch of `steps` blocks https://github.com/firehydrant/terraform-provider-firehydrant/blob/main/examples/runbooks.tf#L2

Reply:
The `steps` naming from the runbook example is weird (`step` makes more sense) but that's [normal Terraform usage](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_firewall) if you bury the membership inside the team and don't specify them as separate resources.

Reply:
We do the same for task_list_items https://github.com/firehydrant/terraform-provider-firehydrant/blob/main/docs/resources/task_list.md#L26 as well. I honestly didn't know if it was plural just to let you know hey you can have more than one block of these.